File: 백태균/week01/12928_약수의_합.java
class Solution {
    public int solution(int n) {
        int answer = 0;

        for (int i=1; i*i<=n; i++) {
            if (n % i == 0) {
                answer += i;  // 작은 약수

                if (i != n / i) {
                    answer += n / i;  // 쌍이 되는 큰 약수 (중복 방지)
                }
            }
        }

        return answer;
    }
}

File: 김광현/week01/12928_약수의_합.py
# 문제: 약수의 합
# URL: https://school.programmers.co.kr/learn/courses/30/lessons/12928

def solution(n):
    divisors = []
    for i in range(n):
        if n % (i + 1) == 0:
            divisors.append(i + 1)
    answer = sum(divisors)
    return answer